object-store-client-play-26/src/main/scala/uk/gov/hmrc/objectstore/client/play/PlayObjectStoreWrites.scala

```diff
@@ -34,26 +32,48 @@ trait PlayObjectStoreWrites {
     new ObjectStoreWrite[Source[ByteString, akka.NotUsed], Request] {
       override def write(body: Source[ByteString, NotUsed]): Request = {
         val tempFile = SingletonTemporaryFileCreator.create()
-        body.runWith(FileIO.toPath(tempFile.path)).map { _ =>
 
+        val (uploadFinished, md5Finished) =
+          broadcast(body, FileIO.toPath(tempFile.path), Md5Hash.md5HashSink)
```

Comment:
It may be because I'm not overly-familiar with Akka Streams, but the method named `broadcast` is a bit confusing.
Something like `parallelSinks` might be better... although I'm still not really convinced by that either :)

Reply:
I think it's standard terminology (same as a broadcast to a load of client sockets)